website/profile/views.py

```diff
@@ -57,15 +56,14 @@ def get_public_projects(uid=None, user=None):
 def get_public_components(uid=None, user=None):
     user = user or User.load(uid)
     # TODO: This should use User.visible_contributor_to?
-
+    # TODO: Should this be limited for users with many projects / components?
```

Comment:
Good questions!

`website.profile.utils.get_projects(...)` was only ever used to get a count, so the list call was totally unnecessary. There, the return value of the function could just be a queryset and there was a big potential performance gain.

By contrast, `get_public_projects` and `get_public_components` are API endpoints designed to return all entries, and they pass the results to `render_nodes` (which immediately turns the result into a list regardless).   Either way, `get_public_projects` actually does convert the result to a list first, in line 53- the functions work the same, just with different syntax.